src/main/java/com/sap/sgs/phosphor/fosstars/nvd/data/AbstractCpeUri.java

```diff
@@ -0,0 +1,70 @@
+package com.sap.sgs.phosphor.fosstars.nvd.data;
+
+import java.util.Objects;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.Pair;
+
+/**
+ * An abstract class for CPE URIs.
+ */
+public abstract class AbstractCpeUri implements CpeUri {
+
+  /**
+   * The vendor of the project.
+   */
+  private final String vendor;
+
+  /**
+   * The product of the project.
+   */
+  private final String product;
+
+  /**
+   * Public constructor.
+   */
+  AbstractCpeUri(String cpeUri, int vendorIndex, int productIndex, String pattern) {
+    Objects.requireNonNull(cpeUri, "The uri cannot be null");
+    Pair<String, String> cpe = extractCpe(cpeUri, pattern, vendorIndex, productIndex);
```

Comment:
I think you could just move the content of `extractCpe` method to the constructor. That would make it shorter.

Reply:
If `(splitContent.length > productIndex)` returns `false`
then `vendor` and `product` could at least have `StringUtils.EMPTY`. In constructor it would look messy.
What do you think?